Brokerages/Oanda/OandaBrokerage.cs

```diff
@@ -0,0 +1,1052 @@
+        /// <param name="requestParams">the parameters to use in the request</param>
+        /// <returns>PostOrderResponse with details of the results (throws if if fails)</returns>
+        public PostOrderResponse PostOrderAsync(Dictionary<string, string> requestParams)
+        {
+            var requestString = EndpointResolver.ResolveEndpoint(OandaEnvironment, Server.Account) + "accounts/" + AccountId + "/orders";
+            return MakeRequestWithBody<PostOrderResponse>(requestString, "POST", requestParams);
+        }
+
+        /// <summary>
+        /// Retrieves the list of open orders belonging to the account
+        /// </summary>
+        /// <param name="requestParams">optional additional parameters for the request (name, value pairs)</param>
+        /// <returns>List of Order objects (or empty list, if no orders)</returns>
+        public List<DataType.Order> GetOrderList(Dictionary<string, string> requestParams = null)
+        {
+            var requestString = EndpointResolver.ResolveEndpoint(OandaEnvironment, Server.Account) + "accounts/" + AccountId + "/orders";
+            var ordersResponse = MakeRequest<OrdersResponse>(requestString, "GET", requestParams);
+            var orders = new List<DataType.Order>();
+            orders.AddRange(ordersResponse.orders);
+            return orders;
+        }
+
+        /// <summary>
+        /// Cancels the order with the specified ID
+        /// </summary>
+        /// <param name="order">The order to cancel</param>
+        /// <returns>True if the request was made for the order to be canceled, false otherwise</returns>
+        public override bool CancelOrder(Order order)
+        {
+            Log.Trace("OandaBrokerage.CancelOrder(): " + order);
+            
+            if (!order.BrokerId.Any())
+            {
+                Log.Trace("OandaBrokerage.CancelOrder(): Unable to cancel order without BrokerId.");
+                return false;
+            }
+
+            foreach (var orderId in order.BrokerId)
+            {
+                CancelOrderAsync(orderId);
+                OnOrderEvent(new OrderEvent(order, "Oanda Cancel Order Event") { Status = OrderStatus.Canceled });
+            }
+
+            return true;
+        }
+
+        private void CancelOrderAsync(long orderId)
+        {
+            var requestString = EndpointResolver.ResolveEndpoint(OandaEnvironment, Server.Account) + "accounts/" + AccountId + "/orders/" + orderId;
+            var response = MakeRequestAsync<Order>(requestString, "DELETE");
+        }
+        
+
+        /// <summary>
+        /// Verify we have a user session; or refresh the access token.
+        /// </summary>
+        public bool RefreshSession()
+        {
+            var raw = "";
+            bool success;
+            lock (_lockAccessCredentials)
+            {
+                //Create the client for connection:
+                var client = new RestClient("https://www.quantconnect.com/terminal/");
```

Comment:
I'm open to suggestions on a different means of persisting access tokens. we did this for tradier, but it looks like we'll need to do this for almost everyone. also, it makes running locally difficult. no action required here for now, but something for those following to think about (and offer suggestions :) )